src/breadcrumb-group/item/item.tsx

```diff
@@ -1,19 +1,117 @@
+import PopoverBody from '../../popover/body';
+import Portal from '../../internal/components/portal';
+import popoverStyles from '../../popover/styles.css.js';
+import { useContainerQuery } from '../../internal/hooks/container-queries';
+import { useMergeRefs } from '../../internal/hooks/use-merge-refs';
+
+type BreadcrumbItemWithPopoverProps<T extends BreadcrumbGroupProps.Item> =
+  React.AnchorHTMLAttributes<HTMLAnchorElement> & {
+    item: T;
+  };
+
+const BreadcrumbItemWithPopover = <T extends BreadcrumbGroupProps.Item>({
+  item,
+  ...anchorAttributes
+}: BreadcrumbItemWithPopoverProps<T>) => {
+  const focusVisible = useFocusVisible();
+  const [textTruncated, setTextTruncated] = useState(false);
+  const [openPopover, setOpenPopover] = useState(false);
+  const trackRef = useRef<HTMLElement>(null);
+  const [textWidth, textRef] = useContainerQuery(rect => rect.width, []);
+  const mergedRef = useMergeRefs(trackRef, textRef);
+  const virtualTextRef = useRef<HTMLElement>(null);
+
+  const popoverContent = (
+    <Transition in={openPopover}>
+      {() => (
+        <PopoverContainer
+          trackRef={trackRef}
+          size="small"
+          fixedWidth={false}
+          position="bottom"
+          arrow={position => (
+            <div className={clsx(popoverStyles.arrow, popoverStyles[`arrow-position-${position}`])}>
+              <div className={popoverStyles['arrow-outer']} />
+              <div className={popoverStyles['arrow-inner']} />
+            </div>
+          )}
+        >
+          <PopoverBody dismissButton={false} dismissAriaLabel={undefined} onDismiss={() => {}} header={undefined}>
+            {item.text}
+          </PopoverBody>
+        </PopoverContainer>
+      )}
+    </Transition>
+  );
+
+  useEffect(() => {
+    if (textWidth && virtualTextRef && virtualTextRef.current) {
+      if (Math.round(virtualTextRef.current.clientWidth) > Math.round(textWidth)) {
+        setTextTruncated(true);
+      }
+    } else {
+      setTextTruncated(false);
+    }
+  }, [textWidth, virtualTextRef]);
+
+  useEffect(() => {
+    const onKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        setOpenPopover(false);
+      }
+    };
+    if (openPopover) {
+      document.addEventListener('keydown', onKeyDown);
+    }
+    return () => {
+      document.removeEventListener('keydown', onKeyDown);
+    };
+  }, [openPopover]);
+
+  return (
+    <>
+      <a
+        {...focusVisible}
+        {...anchorAttributes}
+        onFocus={() => textTruncated && setOpenPopover(true)}
+        onBlur={() => textTruncated && setOpenPopover(false)}
+        onMouseEnter={() => textTruncated && setOpenPopover(true)}
+        onMouseLeave={() => textTruncated && setOpenPopover(false)}
+      >
+        <span className={styles.text} ref={mergedRef}>
+          {item.text}
+        </span>
+      </a>
+      <span className={styles['virtual-item']} ref={virtualTextRef}>
```

Comment:
should we move this inside `<a>` to inherit text styles (and have the same width)?

Reply:
Great point! Updated in the new revision.